src/Stack/Setup.hs

```diff
@@ -609,6 +618,33 @@ getGhcBuild menv = do
         $logDebug ("Using " <> T.pack s <> " GHC build")
         return (CompilerBuildSpecialized s)
 
+-- | Encode an OpenBSD version (like "6.1") into a valid argument for
+-- CompilerBuildSpecialized, so "maj6-min1". Later version numbers are prefixed
+-- with "r".
+-- The result r must be such that "ghc-" ++ r is a valid package name,
+-- as recognized by parsePackageNameFromString.
```

Comment:
Unless things changed, the ghc-build name must be accepted by this parser for package names; that's what the comment here means:
https://github.com/commercialhaskell/stack/blob/ead648fae5ed2369b5d7b2cd28b86dc69b208486/src/Stack/Types/PackageName.hs#L77

So you can replace `maj` and `min` by any other sequence of letters (and maybe numbers). I just went for something mnemonic.
Or you could lift this restriction, but it might be invasive.